### PR TITLE
SONARFLEX-115 Update rule metadata and default quality profile from RSPEC

### PR DESCRIPTION
**Pull request by pynicolas**

### CODE REVIEW

File: sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/FlexRulesDefinition.java [hunk 1]
```diff
@@ -19,22 +19,31 @@
...
+  private static final Set<String> TEMPLATE_RULE_KEYS = new HashSet<>(Arrays.asList("XPath", "CommentRegularExpression"));
 
   @Override
   public void define(Context context) {
     NewRepository repository = context
       .createRepository(CheckList.REPOSITORY_KEY, Flex.KEY)
       .setName(REPOSITORY_NAME);
 
-    new AnnotationBasedRulesDefinition(repository, Flex.KEY).addRuleClasses(false, CheckList.getChecks());
+    // FIXME: with SonarQube 6.7, should use the sonar way profile location as extra parameter
```
Comment (SonarTech) [inline sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/FlexRulesDefinition.java]: Code Smell Code Smell: Take the required action to fix the issue indicated by this comment. ([squid:S1134](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.flex%3Aflex&branch=SONARFLEX-115&issues=AWGuSbKA-_vqYVDUCV73&open=AWGuSbKA-_vqYVDUCV73)

[ISSUE_KEY]: AWGuSbKA-_vqYVDUCV73

Comment (SonarTech) [inline sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/FlexRulesDefinition.java]: Code Smell Code Smell: Take the required action to fix the issue indicated by this comment. ([squid:S1134](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.flex%3Aflex&branch=SONARFLEX-115&issues=AWGuSbKA-_vqYVDUCV73&open=AWGuSbKA-_vqYVDUCV73)

[ISSUE_KEY]: AWGuSbKA-_vqYVDUCV73

File: flex-checks/src/main/resources/org/sonar/l10n/flex/rules/flex/ClassComplexity.html [hunk 2]
```diff
@@ -1,4 +1,5 @@
-<p>
-The cyclomatic complexity of a class should not exceed a defined threshold.
-Complex code can perform poorly and will in any case be difficult to understand and therefore to maintain.
-</p>
+<p>The cyclomatic complexity of a class should not exceed a defined threshold. Complex code can perform poorly and will in any case be difficult to
+understand and therefore to maintain.</p>
+<h2>Deprecated</h2>
+<p>This rule is deprecated, and will eventually be removed.</p>
```
Comment (m-g-sonar) [inline flex-checks/src/main/resources/org/sonar/l10n/flex/rules/flex/ClassComplexity.html]: Deprecated? Are we going to implement cognitive complexity?

File: flex-checks/src/main/resources/org/sonar/l10n/flex/rules/flex/S1312.html [hunk 3]
```diff
@@ -1,20 +1,18 @@
...
-  <li> <code>private</code>: never be accessible outside of its parent class. If another class needs to log something, it should instantiate its own logger.
-  </li><li> <code>static</code>: not be dependent on an instance of a class (an object). When logging something, contextual information can of course be provided in the messages but the logger should be created at class level to prevent creating a logger along with each object.
-</li><li> <code>(final|const)</code>: be created once and only once per class.
-</li></ul>
-
+  <li> <code>private</code>: not accessible outside of their parent classes. If another class needs to log something, it should instantiate its own
+  logger. </li>
+  <li> <code>static</code>: not dependent on an instance of a class (an object). When logging something, contextual information can of course be
+  provided in the messages but the logger should be created at class level to prevent creating a logger along with each object. </li>
+  <li> <code>final</code>: created once and only once per class. </li>
```
Comment (m-g-sonar) [inline flex-checks/src/main/resources/org/sonar/l10n/flex/rules/flex/S1312.html]: const dropped?

File: flex-checks/src/main/resources/org/sonar/l10n/flex/rules/flex/ClassComplexity.html (hunk 2 above)
Comment (pynicolas) [reply flex-checks/src/main/resources/org/sonar/l10n/flex/rules/flex/ClassComplexity.html]: My understanding is that this deprecation is not related to cognitive complexity. This is confirmed by the [SONARFLEX-86](https://jira.sonarsource.com/browse/SONARFLEX-86) which was created by Ann.

File: flex-checks/src/main/resources/org/sonar/l10n/flex/rules/flex/S1312.html (hunk 3 above)
Comment (pynicolas) [reply flex-checks/src/main/resources/org/sonar/l10n/flex/rules/flex/S1312.html]: Fixed in dc85a02

Comment (pynicolas) [reply flex-checks/src/main/resources/org/sonar/l10n/flex/rules/flex/S1312.html]: Thanks!

File: sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/FlexRulesDefinition.java (hunk 1 above)
Comment (SonarTech) [inline sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/FlexRulesDefinition.java]: Code Smell Code Smell: Take the required action to fix the issue indicated by this comment. ([squid:S1134](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.flex%3Aflex&branch=SONARFLEX-115&issues=AWGuSbKA-_vqYVDUCV73&open=AWGuSbKA-_vqYVDUCV73)

[ISSUE_KEY]: AWGuSbKA-_vqYVDUCV73

**Pull request merged**